src/pages/patientView/mutation/PatientViewMutationsDataStore.ts

```diff
@@ -0,0 +1,104 @@
+import {SimpleGetterLazyMobXTableApplicationDataStore} from "../../../shared/lib/ILazyMobXTableApplicationDataStore";
+import {Mutation} from "../../../shared/api/generated/CBioPortalAPI";
+import {action, computed, observable} from "mobx";
+import _ from "lodash";
+import ComplexKeySet from "../../../shared/lib/complexKeyDataStructures/ComplexKeySet";
+
+function mutationMatch(d:Mutation[], id:Mutation) {
+    return d[0].proteinChange === id.proteinChange &&
+        d[0].gene.hugoGeneSymbol === id.gene.hugoGeneSymbol;
+}
+
+function mutationIdKey(m:Mutation) {
+    return `{ "proteinChange": "${m.proteinChange}", "hugoGeneSymbol": "${m.gene.hugoGeneSymbol}" }`;
+}
+
+export default class PatientViewMutationsDataStore extends SimpleGetterLazyMobXTableApplicationDataStore<Mutation[]> {
+    @observable.ref private mouseOverMutation:Readonly<Mutation>|null = null;
+    private selectedMutationsMap = observable.map<Mutation>();
+    @observable private _onlyShowSelectedInTable = false;
+    @observable private _onlyShowSelectedInVAFChart = false;
+
+    public getMouseOverMutation() {
+        return this.mouseOverMutation;
+    }
+
+    public get onlyShowSelectedInTable() {
+        return this._onlyShowSelectedInTable;
+    }
+
+    public get onlyShowSelectedInVAFChart() {
+        return this._onlyShowSelectedInVAFChart;
+    }
+
+    public setMouseOverMutation(m:Readonly<Mutation>|null) {
```

Comment:
action?   i know these only set one thing, but i think it's good for less experienced devs to keep ethic "always make mutating methods actions".